lib/trip_dispatcher.rb

```diff
@@ -90,6 +90,50 @@ def load_trips
       trips
     end
 
+    def available_driver
+      @drivers.each do |driver|
+        if driver.status == :AVAILABLE
+          driver.driver_status_change(driver.status)
+          return driver
+        # else
+        #   raise ArgumentError("No Driver is available!")
+        end
+      end
+    end
+
+    def request_trip(passenger_id)
+      passenger = find_passenger(passenger_id)
+      trip = {}
+
+      trip[:driver] = available_driver
+      trip[:id] = @trips.length + 2
+      trip[:passenger] = passenger
+      trip[:start_time] = Time.now
+      trip[:end_time]= nil
+      trip[:cost] = nil
+      trip[:rating] = nil
+
+      # pass = :passenger
+      new_trip = Trip.new(trip)
+
+      available_driver.add_trip(new_trip)
+
+      passenger.add_trip(new_trip)
+
+
+      return new_trip
+
+    end
+
+    def add_requested_trip
+      trips << request_trip
+      return trips
```

Comment:
You define this method here, but it's never called by `request_trip`. That means the new trip doesn't get recorded!